### ISSUE TITLE
util/log: data race in tests

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#57466]
Saw it [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=2492743&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests_Testrace) on #55713, and I don't think my PR is related.
```
==================
WARNING: DATA RACE
Write at 0x000009cf14b0 by goroutine 1545:
  github.com/cockroachdb/cockroach/pkg/util/log.ApplyConfig()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/flags.go:226 +0x2d4
  github.com/cockroachdb/cockroach/pkg/util/log.ScopeWithoutShowLogs()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:165 +0x864
  github.com/cockroachdb/cockroach/pkg/util/log.Scope()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/test_log_scope.go:72 +0x64
  github.com/cockroachdb/cockroach/pkg/sql_test.TestScatterRandomizeLeases()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scatter_test.go:33 +0xb2
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1123 +0x202

Previous read at 0x000009cf14b0 by goroutine 1791:
  ??()
      -:0 +0xffffffffffffffff
  sync.(*Pool).Get()
      /usr/local/go/src/sync/pool.go:128 +0x44
  fmt.newPrinter()
      /usr/local/go/src/fmt/print.go:137 +0x3e
  fmt.Fprint()
      /usr/local/go/src/fmt/print.go:231 +0x33
  github.com/cockroachdb/redact.reproducePrintf()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/make_format.go:28 +0xbe
  github.com/cockroachdb/redact.safeWrapper.Format()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/wrappers.go:57 +0xd9
  github.com/cockroachdb/redact.(*safeWrapper).Format()
      <autogenerated>:1 +0x2e
  github.com/cockroachdb/redact.(*escapeArg).Format()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/markers_internal_print.go:235 +0x457
  github.com/cockroachdb/redact/internal.(*pp).handleMethods()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/internal/print.go:609 +0x1c7
  github.com/cockroachdb/redact/internal.(*pp).printArgOrig()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/internal/print.go:726 +0x1a4
  github.com/cockroachdb/redact/internal.PrintArg()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/internal/hooks.go:68 +0x786
  github.com/cockroachdb/redact.printArgFn()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/markers_internal_print.go:93 +0x760
  github.com/cockroachdb/redact/internal.(*pp).printArg()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/internal/hooks.go:21 +0x92
  github.com/cockroachdb/redact/internal.(*pp).doPrint()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/internal/print.go:1174 +0xe6
  github.com/cockroachdb/redact/internal.DoPrint()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/internal/hooks.go:71 +0xa6
  github.com/cockroachdb/redact.Fprint()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/redact/markers_print.go:105 +0x76
  github.com/cockroachdb/cockroach/pkg/util/log.renderTagsAsRedactable()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/redact.go:197 +0x270
  github.com/cockroachdb/cockroach/pkg/util/log.MakeEntry()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_entry.go:98 +0x244
  github.com/cockroachdb/cockroach/pkg/util/log.logfDepth()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/channels.go:44 +0x106
  github.com/cockroachdb/cockroach/pkg/util/log.Warningf()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_channels_generated.go:431 +0x297
  github.com/cockroachdb/cockroach/pkg/kv.(*Txn).CleanupOnError()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:618 +0x1f0
  github.com/cockroachdb/cockroach/pkg/sql.cleanupAndFinishOnError()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_fsm.go:472 +0x127
  github.com/cockroachdb/cockroach/pkg/util/fsm.Transitions.apply()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/fsm/fsm.go:106 +0x203
  github.com/cockroachdb/cockroach/pkg/util/fsm.(*Machine).ApplyWithPayload()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/fsm/fsm.go:135 +0xf64
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).close()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:819 +0xd5f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:802 +0x132
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:479 +0x6e
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:481 +0x154
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:621 +0x3a4

Goroutine 1545 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1168 +0x5bb
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1439 +0xa6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1123 +0x202
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1437 +0x612
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1345 +0x3b3
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:32 +0x15b
  main.main()
      _testmain.go:689 +0x271

Goroutine 1791 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:549 +0x1f7
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:275 +0x8d4
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:154 +0x30a
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:589 +0xa84
  github.com/cockroachdb/cockroach/pkg/server.(*sqlServer).startServeSQL.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1964 +0x1a4
  github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:148 +0x11d
```
cc @knz for triage

**Comment by knz** [comment]
fixed by #57454

**Comment by yuzefovich** [comment]
Thanks! Closing as a dup then.
